OpenRA.Game/Map/MapCache.cs

```diff
@@ -174,7 +174,8 @@ void LoadAsyncInternal()
 					if (bitmap == null)
 					{
 						createdPreview = true;
-						bitmap = Minimap.RenderMapPreview(modData.DefaultRules.TileSets[p.Map.Tileset], p.Map, modData.DefaultRules, true);
+						var map = new Map(p.Path);
```

Comment:
The main issue is reading from the filesystem, so a better approach may be to:

1.  Use locks around the innards of the various Filesystem code to avoid race conditions between e.g. mounting packages and querying contents
2.  Make sure that the stream objects returned by the package implementations do not implicitly share state.